### PR TITLE
Syndicate Camera

### PR DESCRIPTION
**Pull request by Reviire**
Adds a new camera for traitor librarians. Costs 15TC.
Anyone you take a picture of will eventually cause a small explosion.

Clusterfack said it'd be cool.

Related - AAAAH how does github even work help

### CODE REVIEW

File: code/modules/paperwork/photography.dm [hunk 1]
```diff
@@ -379,6 +387,24 @@
...
+			if(0)
+				holding = "They are holding \a [I]"
+			else
+				holding += " and \a [I]"
+		item_count++
+
+		if(!mob_detail)
+			mob_detail = "You can see [A] on the photo[A:health < 75 ? " - [A] looks hurt":""].[holding ? " [holding]":"."]. "
+		else
+			mob_detail += "You can also see [A] on the photo[A:health < 75 ? " - [A] looks hurt":""].[holding ? " [holding]":"."]."
```
Comment (unid15) [inline code/modules/paperwork/photography.dm]: You can't modify variables like that in byond, this proc won't do anything unless either mob_detail becomes a higher-level variable (i.e. a camera variable), or it's modified to actually return a value

Comment (Exxion) [reply code/modules/paperwork/photography.dm]: >tfw BYOND will never have explicit pass-by-reference

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -123,6 +123,30 @@
...
+/obj/item/device/camera/syndicate/get_mob_items(var/mob_detail, turf/the_turf, var/mob/living/carbon/A)
+	startexplosion(A)
+	return ..()
+
+
+/obj/item/device/camera/syndicate/proc/startexplosion(atom/target) // Promise this isn't shitcode
+	var/timer = rand(30 SECONDS, 420 SECONDS)
+
+	spawn(round(timer*0.4))
+		to_chat(target, "Things are getting a bit spooky")
```
Comment (ShiftyRail) [inline code/modules/paperwork/photography.dm]: Why wouldn't you add a dot ?
Same thing with the other `to_chat` messages.

**Comment by ShiftyRail** [comment]
Needs a changelog : )
Also if we want to be petty it is `\the [O]` but whatever

File: code/modules/paperwork/photography.dm [hunk 3]
```diff
@@ -123,6 +123,30 @@
...
+
+/obj/item/device/camera/syndicate/proc/startexplosion(atom/target) // Promise this isn't shitcode
+	var/timer = rand(30 SECONDS, 420 SECONDS)
+
+	spawn(round(timer*0.4))
+		to_chat(target, "Things are getting a bit spooky.")
+	spawn(round(timer*0.7))
+		to_chat(target, "It's getting really spooky.")
+	spawn(timer-20)
+		to_chat(target, "<span class ='warning'>Oh shit SPOOKED!</span>")
```
Comment (ghost) [inline code/modules/paperwork/photography.dm:144]: 😕

**Comment by Lokiamis** [comment]
8d4306b3-bf72-40eb-9d91-d0ab9eb69ac2

**Comment by ghost** [comment]
So you can just get a few rolls of camera, nigger a camera console and take pictures of the whole station?

That's shit both from a gameplay perspective and thematically.

**Comment by ghost** [comment]
Is this an obscure anime reference about an anime where the main character can make explosions by taking pictures because of some cult he walked in on and now has to beat up the mega Jew behind the secret society

**Comment by Exxion** [comment]
So:
1. The target knows they're going to explode well ahead of time, but can't do anything about it except avoid people and fragile areas
2. The explosion is a whopping 3x3 light impact, which is not even lethal unless the target is a total dumbass and stands beside a welder tank (unless you count death from bleeding)
3. You have no idea where the target is going to be after the preposterously long timer, unless you've captured them, in which case there are much better options already
4. You out yourself as traitor immediately because it's job-specific

It sounds like this item sucks both for the user and the target

EDIT: I didn't notice that it works on everyone in the picture. That's not as bad as I was thinking, but it still doesn't seem very good.

**Comment by ghost** [comment]
You can also just take 5 pictures of your target and they will all stack.

**Comment by Exxion** [comment]
Considering the timer lasts anywhere between half a minute and *seven minutes*, that's barely useful
Just means you grab a few bruise packs when you get the first message

**Comment by Reviire** [comment]
Well it's more about the idea at this point. I could remove the messages, I just thought not being informed about exploding would be a bit much.  I could reduce the difference between the explosion times, so you could stack it up easier. Add some way to change the timer length as well.

I don't know how i'd be able to fix niggering through a cam console, a distance check between the camera and where the picture is being taken? 

I dunno how I could make the explosions stronger. I tried heavy explosions and they'd cause breaches, which seemed a bit too good.

**Comment by Blithering** [comment]
does it also change everyone's race in the pictures to skellington?

**Comment by Aurx** [comment]
So it requires you to blatantly advertise to your target(s) that you're a traitor, won't kill anyone reliably even if spammed, gives them nigh unto forever to hunt you down and lynch your ass, and can be trivially counter-played by gargling bicard and standing in dorms?  

About all this seems useful for is damaging immulsions and camera-console hiding faggotry.

**Comment by Kurfursten** [comment]
Also the messages are vague so it rewards powergamers and hurts newfags who have no idea what "feeling spooky" means

**Comment by ShiftyRail** [comment]
I must join the others and say I don't like that as a feature.

We already have the Chef's special sauce for that kind of playstyle ; it is extremely punishing the first time, then trivially easy to notice (even moreso with your camera) and to lynch the chef accordingly.
Your item is a somewhat weaker version of that.

Besides, I find the text messages to be rather bad and out of place compared to the rest of the game.

A better idea - but harder to code - would be to allow the syndie cam to track the people it took pictures off.
In emergencies, it could be a flash for close combaat.

**Comment by Reviire** [comment]
I'm going to work on changing it. I was thinking of going with some sort of real-time picture gimmick. Either allowing it to track people it took pictures of, or allowing the person who took the picture to interact with objects within the picture.

**Comment by SonixApache** [comment]
How about if it made photos that explode (0,0,1) when viewed?
Put a bomb in the middle of the photo for maximum memes.

**Comment by Gunosaguki** [comment]
And people wonder why some people are so afraid of getting their picture taken

**Comment by Pathid** [comment]
Idea: Pickpocketing (or reverse-pickpocketing) through the photo. Also soul stones should function through it.

**Comment by PJB3005** [comment]
Is this going to be fixed or can I close?

**Pull request closed without merging**